pkg/graph/resolvers/model_plan_phase.go

```diff
@@ -138,42 +143,107 @@ func TrySendEmailForPhaseSuggestion(
-	// Get the model leads for the model plan
+	// Get the collaborators for the model plan
 	planCollaborators, err := store.PlanCollaboratorGetByModelPlanID(logger, modelPlanID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get plan collaborators: %w", err)
 	}
 
 	logger.Info("Fetched collaborators", zap.Any("collaborators", planCollaborators))
 
-	var modelLeadEmails []string
+	var emailRecipients []string
 	for _, collaborator := range planCollaborators {
 		teamRoles := models.ConvertEnums[models.TeamRole](collaborator.TeamRoles)
 		logger.Info("Converted roles", zap.Any("roles", teamRoles))
 
-		for _, role := range teamRoles {
-			if role == models.TeamRoleModelLead {
-				account, accountErr := collaborator.UserAccount(ctx)
-				if accountErr != nil {
-					logger.Error("Failed to get user account", zap.Error(accountErr))
-					continue
-				}
-				logger.Info("Fetched user account", zap.String("email", account.Email))
-				modelLeadEmails = append(modelLeadEmails, account.Email)
-				break
-			}
+		acct, accErr := collaborator.UserAccount(ctx)
+		if accErr != nil {
+			logger.Warn("failed to load user account", zap.Error(accErr), zap.String("userID", collaborator.UserID.String()))
+			continue
+		}
+		if acct.Email == "" {
+			continue
+		}
+
+		if lo.Contains(teamRoles, models.TeamRoleModelLead) {
+			emailRecipients = append(emailRecipients, acct.Email)
+			continue
+		}
+
+		// Non-lead: include only if opted in for incorrect model status email
+		pref, prefErr := UserNotificationPreferencesGetByUserID(ctx, acct.ID)
+		if prefErr != nil {
+			logger.Warn("unable to get user notification preferences", zap.Error(prefErr), zap.String("userID", acct.ID.String()))
+			continue
+		}
+		if pref != nil && pref.IncorrectModelStatus.SendEmail() {
+			emailRecipients = append(emailRecipients, acct.Email)
+		}
+	}
+
+	logger.Info("Final email recipients", zap.Strings("emails", emailRecipients))
+	return emailRecipients, nil
+}
+
+// GetCollaboratorUUIDsForModelStatusAlert returns the UUIDs of all collaborators for a given model plan
+func GetCollaboratorUUIDsForModelStatusAlert(
+	ctx context.Context,
+	logger *zap.Logger,
+	store *storage.Store,
+	modelPlanID uuid.UUID,
+) ([]uuid.UUID, error) {
+	planCollaborators, err := store.PlanCollaboratorGetByModelPlanID(logger, modelPlanID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get plan collaborators: %w", err)
+	}
+
+	var uuids []uuid.UUID
+	for _, collaborator := range planCollaborators {
+		uuids = append(uuids, collaborator.UserID)
+	}
+
+	logger.Info("Final collaborator UUIDs", zap.Any("uuids", uuids))
+	return uuids, nil
+}
+
+func sendInAppNotificationForPhaseSuggestion(
+	ctx context.Context,
+	store *storage.Store,
+	logger *zap.Logger,
+	receiverIDs []uuid.UUID,
+	modelPlanID uuid.UUID,
+	phaseSuggestion *model.PhaseSuggestion,
+	modelPlan *models.ModelPlan,
+) error {
+	//Future Enhancement use the dataloader to get user preferences and remove the getPreferencesFunc
```

Comment:
Note, I think that we have access to the data loaders now? Would it be worthwhile to add that on?

Reply:
added